### PR TITLE
Minor improvements to `quantum_info` documentation

### LINKED ISSUE #898: `circuit.with_noise` bug
I was following the [documentation example](https://qibo.science/qibo/stable/api-reference/qibo.html#qibo.models.circuit.Circuit.with_noise) for building a noisy circuit. 
By running the following:

```python
import numpy as np
import matplotlib.pyplot as plt

from qibo import gates, set_backend
from qibo.models import Circuit

set_backend('numpy')

nqubits = 1
noise_map = {0: (0.1, 0.0, 0.2)}

c = Circuit(nqubits)
c.add(gates.RZ(q=0, theta=0))
c.add(gates.RZ(q=0, theta=0))
c.add(gates.RY(q=0, theta=0))
c.add(gates.RY(q=0, theta=0))
c.add(gates.M(0))

noisy_c = c.with_noise(noise_map)
```

I get this:

```
File "/home/matteo/Documents/PhD/qibo/src/qibo/models/circuit.py", line 505, in <listcomp>
if q in noise_map and sum([row[1] for row in noise_map[q]]) > 0:
TypeError: 'float' object is not subscriptable
```

I think is related to the new way to use the noise channels.

### PR DESCRIPTION
**Pull request by renatomello**
This implements minor improvements in the documentation of the `quantum_info` module, and also closes #898 

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Review by vodovozovaliza — APPROVED** [review]

**Review by MatteoRobbiati — COMMENTED** [review]
Thank you for this.
Just a small comment follows.

File: src/qibo/models/circuit.py [hunk 1]
```diff
@@ -449,10 +449,9 @@ def with_noise(self, noise_map: NoiseMapType):
...
 
         Args:
-            noise_map (dict): Dictionary that maps qubit ids to noise
-                probabilities (px, py, pz).
-                If a tuple of probabilities (px, py, pz) is given instead of
-                a dictionary, then the same probabilities will be used for all
+            noise_map (dict): list of tuples :math:`(P_{k}, p_{k})`, where
+                :math:`P_{k}` is a ``str`` representing the :math:`k`-th Pauli
+                operator, and :math:`p_{k}` is the associated probability.
                 qubits.
```
Comment (MatteoRobbiati) [inline src/qibo/models/circuit.py:455]: Maybe here the sentence is not finished.

**Review by MatteoRobbiati — COMMENTED** [review]
Thank you for this.
Just a small comment follows.

File: src/qibo/models/circuit.py (hunk 1 above)
Comment (renatomello) [reply src/qibo/models/circuit.py]: Thank you. Already fixed it.

**Review by MatteoRobbiati — APPROVED** [review]
Thanks.

**Pull request merged**